vllm/config/compilation.py

```diff
@@ -687,14 +707,15 @@ def splitting_ops_contain_attention(self) -> bool:
 
     def is_attention_compiled_piecewise(self) -> bool:
         use_fx_graph_piecewise_compilation = (
-            self.level == CompilationLevel.PIECEWISE
+            self.level == CompilationMode.VLLM_COMPILE
             and self.splitting_ops_contain_attention()
         )
 
         inductor_used = (
-            self.level == CompilationLevel.PIECEWISE and self.use_inductor
+            self.level == CompilationMode.VLLM_COMPILE and self.use_inductor
         ) or (
-            self.level >= CompilationLevel.DYNAMO_AS_IS and self.backend == "inductor"
+            self.level >= CompilationMode.STOCK_TORCH_COMPILE
+            and self.backend == "inductor"
         )
```

Comment:
I think after #26113, we can simplify this (might not even need a variable):

```suggestion
inductor_used = self.backend == "inductor"
```

We should also make sure backend is "" or "eager" if mode=NONE.